flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/JobMasterPartitionTrackerImpl.java

```diff
@@ -95,27 +110,31 @@ public void stopTrackingAndReleasePartitions(Collection<ResultPartitionID> resul
     }
 
     @Override
-    public void stopTrackingAndReleasePartitionsFor(ResourceID producingTaskExecutorId) {
-        Preconditions.checkNotNull(producingTaskExecutorId);
-
-        Collection<ResultPartitionDeploymentDescriptor> resultPartitionIds =
-                CollectionUtil.project(
-                        stopTrackingPartitionsFor(producingTaskExecutorId),
-                        PartitionTrackerEntry::getMetaInfo);
+    public void stopTrackingAndReleaseOrPromotePartitions(
```

Comment:
Maybe we can add a method `stopTrackingAndHandlePartitions(...)` and reuse it in both`stopTrackingAndReleasePartitions(...)` and `stopTrackingAndReleaseOrPromotePartitions(...)`?
```
    public void stopTrackingAndHandlePartitions(
            Collection<ResultPartitionID> resultPartitionIds,
            BiConsumer<ResourceID, Collection<ResultPartitionDeploymentDescriptor>>
                    partitionHandler) {
        Preconditions.checkNotNull(resultPartitionIds);

        // stop tracking partitions to handle and group them by task executor ID
        Map<ResourceID, List<ResultPartitionDeploymentDescriptor>> partitionsToReleaseByResourceId =
                stopTrackingPartitions(resultPartitionIds).stream()
                        .collect(
                                Collectors.groupingBy(
                                        PartitionTrackerEntry::getKey,
                                        Collectors.mapping(
                                                PartitionTrackerEntry::getMetaInfo, toList())));

        partitionsToReleaseByResourceId.forEach(partitionHandler);
    }
```

Reply:
cool ~~